### PR TITLE
[formrecognizer] model_id param validation

### PR DESCRIPTION
**Pull request by kristapratico**
Item on arch board issues #11283 

Original suggestion was to to enforce that user passes guid for methods that accept `model_id` parameter. Based on architect guidance we will not validate for this, but will validate that model_id is not None or an empty string. 

Reasoning is that service could eventually allow users to name their models with "real" names and not just server generated guids. It's reasonable to check for None/empty string since this will likely never be a valid model_id and prevents a 404 error in the path URL.

### CODE REVIEW

**Comment by kristapratico** [comment]
@maririos FYI since we talked about this

**Comment by maririos** [comment]
ohh interesting. 
In .NET we do validate that we can convert the string into a GUID, because currently the service expects a UUID. It is worth to checking this in Python?

FYI @samvaity as she was looking into consistency across languages for this item

**Comment by samvaity** [comment]
>In .NET we do validate that we can convert the string into a GUID, because currently the service expects a UUID. It is worth to checking this in Python?

I think we should validate if it is a valid UUID. Especially since the service does not validate that eagerly and will result in a request failure.
We can save users time and cost by failing early on client-side.
What do you think @kristapratico ?

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py [hunk 1]
```diff
@@ -277,6 +277,9 @@ def begin_recognize_custom_forms(self, model_id, form, **kwargs):
                 :caption: Recognize fields and values from a custom form.
         """
 
+        if not model_id:
```
Comment (rakshith91) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:280]: Curious - don't we have match conditions for delete operations in form?

**Comment by kristapratico** [comment]
> I think we should validate if it is a valid UUID. Especially since the service does not validate that eagerly and will result in a request failure.
We can save users time and cost by failing early on the client-side.
What do you think @kristapratico ?

I talked to Johan about this and posted the reasoning above:

Reasoning is that service could eventually allow users to name their models with "real" names and not just server generated guids. It's reasonable to check for None/empty string since this will likely never be a valid model_id and prevents a 404 error in the path URL.

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py (hunk 1 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:280]: Can you show me where you see that?

**Comment by samvaity** [comment]
>I talked to Johan about this and posted the reasoning above:
Reasoning is that service could eventually allow users to name their models with "real" names and not just server generated guids. It's reasonable to check for None/empty string since this will likely never be a valid model_id and prevents a 404 error in the path URL.

Not sure how other languages are handling this currently, but Java will always first convert the user-provided string to a UUID, using `fromString()` from UUID's lib class to pass it down to the service. If the string provided to `fromString()` does not conform to UUID specifications it will throw an `IllegalArgumentException`.

**Comment by kristapratico** [comment]
> Not sure how other languages are handling this currently, but Java will always first convert the user-provided string to a UUID, using fromString() from UUID's lib class to pass it down to the service. If the string provided to fromString() does not conform to UUID specifications it will throw an IllegalArgumentException.

But if the service eventually allows the user to name the model, e.g. "myInvoiceModel", won't this fail with `IllegalArgumentException`? This is what Johan wants to avoid

**Comment by samvaity** [comment]
>But if the service eventually allows the user to name the model, e.g. "myInvoiceModel", won't this fail with IllegalArgumentException? This is what Johan wants to avoid

To allow this the service will probably have to start accepting Strings in their API's and that's when we (Java) can plan to remove the `UUID.fromString()`. Currently, with the service expecting the type UUID, there is no way of escaping converting the user-provided string to UUID.

**Pull request merged**